digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadLinkService.scala

```diff
@@ -921,7 +921,12 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
     val diagonal = Vector3d(10, 10, 0)
 
     val roadLinks =
-      if (user.isOperator()) getVVHRoadLinks(BoundingRectangle(point - diagonal, point + diagonal))
+      if (user.isOperator()) {
+        getVVHRoadLinks(BoundingRectangle(point - diagonal, point + diagonal)).filter(_.administrativeClass != State)
+      }
+      else if (user.isMunicipalityMaintainer()) {
+        getVVHRoadLinks(BoundingRectangle(point - diagonal, point + diagonal), user.configuration.authorizedMunicipalities).filter(_.administrativeClass.value != State)
```

Comment:
.filter(_.administrativeClass.value != State) kohta lähenee duplikaattia, ei vakavaa mutta sä voisit siirtää filteroidin itse roadLinks muuttujalle. Vaikkapa roadlinkWithCorrectArea ja sitten val stateRoadFillteredAway = roadlinkWithCorrectArea..filter(_.administrativeClass.value != State)  jota sitten käytetään tästä eteen päin. Vaatimuthan oli, että kukaan ei saa lisätä valtion tielle, joten ei ole väliä millä oikeudellisille linkeille tehdään filter

Reply:
ehkä tätä ei sittenkään kannata muokata kun tossa lopussa on toi else